pyogrio/tests/conftest.py

```diff
@@ -18,14 +18,35 @@ def pytest_report_header(config):
     )
 
 
+def prepare_testfile(testfile_path, dst_dir, ext):
+    if ext == testfile_path.suffix:
+        return testfile_path
+    else:
+        dst_path = dst_dir / f"{testfile_path.stem}{ext}"
+        if dst_path.exists():
+            return dst_path
+        gdf = pyogrio.read_dataframe(testfile_path)
+        pyogrio.write_dataframe(gdf, dst_path)
+        return dst_path
+
+
 @pytest.fixture(scope="session")
 def data_dir():
     return _data_dir
 
 
-@pytest.fixture(scope="session")
-def naturalearth_lowres():
-    return _data_dir / Path("naturalearth_lowres/naturalearth_lowres.shp")
+@pytest.fixture(scope="function")
+def naturalearth_lowres(tmp_path, request):
+    ext = request.param \
+            if ("param" in dir(request) and request.param is not None) else ".shp"
```

Comment:
is `request` a `dict` or a class?

If the former `request.get('param', '.shp')` and if the latter `getattr(request, 'param', '.shp')`

Reply:
It's a class. That's indeed a lot cleaner. It doesn't support the case where the param is passed in as None, but maybe that's not worth extra complexity anyway... Tnx!